pkg/kubelet/cm/dra/resources.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+// resource contains resource attributes required
+// to prepare and unprepare the resource
+type resource struct {
+	sync.Mutex
+
+	// name of the DRA driver
+	driverName string
+
+	// claimUID is an UID of the resource claim
+	claimUID types.UID
+
+	// claimName is a name of the resource claim
+	claimName string
+
+	// namespace is a claim namespace
+	namespace string
+
+	// podUIDs is a set of pod UIDs that reference a resource
+	podUIDs sets.String
+
+	// cdiDevice is a list of CDI devices returned by the
+	// GRPC API call NodePrepareResource
+	cdiDevice []string
+
+	// annotations is a list of container annotations associated with
+	// a prepared resource
+	annotations []kubecontainer.Annotation
+}
+
+func (res *resource) addPodUID(podUID types.UID) {
```

Comment:
Can we call this `addPodReference()` since its about adding a pod UID that references the resource claim. We can leave out the UID in the function name since its clear from the argument that we pass a UID.

Reply:
Resolved